base/subarray.jl

```diff
@@ -134,6 +134,12 @@ function view(A::AbstractArray, I::Vararg{Any,N}) where {N}
     unsafe_view(_maybe_reshape_parent(A, index_ndims(J...)), J...)
 end
 
+# Ranges tend to be compact & immutable
+function view(r1::AbstractRange, r2::AbstractRange)
+    @_propagate_inbounds_meta
+    getindex(r1, r2)
```

Comment:
Given that we don't have a generic `getindex(::AbstractRange, ::AbstractRange)`, this isn't guaranteed to always return an `AbstractRange` back.  E.g.,

```
julia> (1:3)[StepRangeLen{Int, Int, Int}(1, 1, 3)]
3-element Array{Int64,1}:
 1
 2
 3
```

Maybe we should also define a `getindex(::AbstractRange, ::AbstractRange)` error method to guarantee this property?  Or add a `::AbstractRange` assertion here?

Reply:
Hmm... this is unfortunate.

For the moment I'll add an assertion locally here, if that seems reasonable?